adaway-linux.sh

```diff
@@ -61,20 +61,32 @@ while read src; do
         # - remove additional localhost entries possibly picked up from sources
         # - remove remaining comments
         # - split all entries with one tab
-wget  "${src}"  -nv --show-progress -L -O - \
+        #which curl;
+        if type curl 2>/dev/null > /dev/null; then
+          curl --progress-bar -L "${src}" \
             | sed 's/\r/\n/' \
             | sed 's/^\s\+//' \
             | sed 's/^127\.0\.0\.1/0.0.0.0/' \
             | grep '^0\.0\.0\.0' \
             | grep -v '\slocalhost\s*' \
             | sed 's/\s*\#.*//g' \
             | sed 's/\s\+/\t/g' \
-            >> "${TMPDIR}hosts.downloaded"
-    else
+            >> "${TMPDIR}hosts.downloaded";
+        else
+          wget  "${src}" -nv --show-progress -L -O - \
+            | sed 's/\r/\n/' \
+            | sed 's/^\s\+//' \
+            | sed 's/^127\.0\.0\.1/0.0.0.0/' \
+            | grep '^0\.0\.0\.0' \
+            | grep -v '\slocalhost\s*' \
+            | sed 's/\s*\#.*//g' \
+            | sed 's/\s\+/\t/g' \
+            >> "${TMPDIR}hosts.downloaded";
+        fi
+      else
```

Comment:
I like the idea of supporting both, curl and wget, but we shouldn't duplicate code. This code is even the most important code in the entire project :wink:
What about setting a variable like `DOWNLOAD_CMD` to either `wget -nv --show-progress -L -O -` or `curl --progress-bar -L` and then using `eval "${DOWNLOAD_CMD} ${src}"`?